### ISSUE TITLE
RuntimeError: vmap: It looks like you're calling .item() on a Tensor.

### ISSUE THREAD

**Issue by riteshkhrn** [pytorch/pytorch#102109]
Code:
```
def foo(a, ind, len):
    ind2 = ind + len
    return a[ind:ind2]

torch.vmap(foo, in_dims=(0, 0, None))(a, c, 2)
```
RuntimeError: vmap: It looks like you're calling .item() on a Tensor. We don't support vmap over calling .item() on a Tensor, please try to rewrite what you're doing with other operations. If error is occurring somewhere inside PyTorch internals, please file a bug report.

**Comment by zou3519** [comment]
@mra-h are you hitting the same exact thing where you're trying to vmap over an indexing operation like in the above repro?

**Comment by mra-h** [comment]
Yes I am trying to create a skew matrix from a 3D tensor which has been vmapped from a batch. 

Something like this

````
  return torch.tensor([[0.0, -w[2], w[1]],
                       [w[2], 0.0, -w[0]],
                       [-w[1], w[0], 0.0]])
````

**Comment by mra-h** [comment]
I think this is something important to add support to. In jax, this code is allowed with vmap

**Comment by zou3519** [comment]
@mra-h your code is a separate issue from the one reported here, so I'll make a new one for it

**Comment by jacksonloper** [comment]
If you absolutely need to do this here is one workaround.  Not as efficient as slicing probably.  

```
import torch
def foo(a,ind,sz):
    idxs=torch.arange(sz)+ind
    return torch.index_select(a,0,idxs)

a=torch.arange(20).reshape((2,10))
c=torch.tensor([2,5])
print(torch.vmap(foo,in_dims=(0,0,None))(a,c,2)) # this is inside a vmap
print([foo(suba,subc,2) for (suba,subc) in zip(a,c)]) # but this is the same content
```

**Comment by fredfurst** [comment]
I'm unclear on whether this applies. Here´s my use case: a dynamical system that needs to be integrated over time iteratively.
>     delH_delphi1 = lambda phi1,phi2,J1,J2: -2*J1*J2*alpha*torch.sin(2*phi1 - 2*phi2) - 2*J1*J2**1.5*beta*torch.sin(2*phi1 - 2*phi2)
>     delH_delphi2 = lambda phi1,phi2,J1,J2: 2*J1*J2*alpha*torch.sin(2*phi1 - 2*phi2) + 2*J1*J2**1.5*beta*torch.sin(2*phi1 - 2*phi2)
>     delH_delJ1 = lambda phi1,phi2,J1,J2: -2*J1 + J2*alpha*torch.cos(2*phi1 - 2*phi2) - 3*J2 + J2**1.5*beta*torch.cos(2*phi1 - 2*phi2) + 1
>     delH_delJ2 = lambda phi1,phi2,J1,J2: 1.5*J1*J2**0.5*beta*torch.cos(2*phi1 - 2*phi2) + J1*alpha*torch.cos(2*phi1 - 2*phi2) - 3*J1 + 2*J2 + 1
and then
>         noi = number_of_iterations
>         def sistema_dinamico(lista):
>             phi1 = lista[0]
>             phi2 = lista[1]
>             J1 = lista[2]
>             J2 = lista[3]
>             lista2 = torch.tensor(
>                 (phi1 + dt*delH_delJ1(phi1,phi2,J1,J2),
>                 phi2 + dt*delH_delJ2(phi1,phi2,J1,J2),
>                 J1 - dt*delH_delphi1(phi1,phi2,J1,J2),
>                 J2 - dt*delH_delphi2(phi1,phi2,J1,J2),)
>             )
>             return lista2
>         phiJ[:][1:noi+1] = torch.vmap(sistema_dinamico)(phiJ[:][0:noi])
Seems like this should be supported.
